components/cmake/modules/FindNETCDF.cmake

```diff
@@ -0,0 +1,66 @@
+# - Try to find Netcdf
+#
+# Placeholder code until spio gets a proper config.cmake
+#
+# Once done, this will define:
+#
+#   The "netcdf" target
+#
+
+# Pnetcdf is optional, and only if not running serial
+if (NOT MPILIB STREQUAL mpi-serial)
+  if (PNETCDF_PATH)
+    find_library(pnetcdf_lib pnetcdf REQUIRED PATHS ${PNETCDF_PATH}/lib)
+    find_path (pnetcdf_incdir pnetcdf.h REQUIRED PATHS ${PNETCDF_PATH}/include)
+  endif()
+endif()
+
+if (NETCDF_C_PATH)
+  # Sanity checks
+  if (NOT NETCDF_FORTRAN_PATH)
+    message(FATAL_ERROR "NETCDF_C_PATH specified without NETCDF_FORTRAN_PATH")
+  endif()
+  if (NOT EXISTS ${NETCDF_C_PATH}/lib AND NOT EXISTS ${NETCDF_C_PATH}/lib64)
+    message(FATAL_ERROR "NETCDF_C_PATH does not contain a lib or lib64 directory")
+  endif ()
+  if (NOT EXISTS ${NETCDF_FORTRAN_PATH}/lib AND NOT EXISTS ${NETCDF_FORTRAN_PATH}/lib64)
+    message(FATAL_ERROR "NETCDF_FORTRAN_PATH does not contain a lib or lib64 directory")
+  endif ()
+
+  # Find the libraries
+  find_library(netcdf_c_lib netcdf  REQUIRED PATHS ${NETCDF_C_PATH}/lib ${NETCDF_C_PATH}/lib64)
+  find_library(netcdf_f_lib netcdff REQUIRED PATHS ${NETCDF_FORTRAN_PATH}/lib ${NETCDF_FORTRAN_PATH}/lib64)
+  find_path (netcdf_c_incdir netcdf.h REQUIRED PATHS ${NETCDF_C_PATH}/include)
+  find_path (netcdf_f_incdir netcdf.inc REQUIRED PATHS ${NETCDF_FORTRAN_PATH}/include)
+
+elseif (NETCDF_FORTRAN_PATH)
+  message(FATAL_ERROR "NETCDF_FORTRAN_PATH specified without NETCDF_C_PATH")
+elseif (NETCDF_PATH)
+
+  # Sanity checks
+  if (NOT EXISTS ${NETCDF_PATH}/lib AND NOT EXISTS ${NETCDF_PATH}/lib64)
+    message(FATAL_ERROR "NETCDF_PATH does not contain a lib or lib64 directory")
+  endif ()
+
+  find_library(netcdf_c_lib netcdf  REQUIRED PATHS ${NETCDF_PATH}/lib ${NETCDF_PATH}/lib64)
+  find_library(netcdf_f_lib netcdff REQUIRED PATHS ${NETCDF_PATH}/lib ${NETCDF_PATH}/lib64)
+  find_path (netcdf_c_incdir netcdf.h REQUIRED PATHS ${NETCDF_PATH}/include)
+  find_path (netcdf_f_incdir netcdf.inc REQUIRED PATHS ${NETCDF_PATH}/include)
+else()
+  message(FATAL_ERROR "NETCDF not found: Define NETCDF_PATH or NETCDF_C_PATH and NETCDF_FORTRAN_PATH in config_machines.xml or config_compilers.xml")
+endif()
+
+set (pnetcdf_lib ${pnetcdf_lib} PARENT_SCOPE)
+set (netcdf_c_lib ${netcdf_c_lib} PARENT_SCOPE)
+set (netcdf_f_lib ${netcdf_f_lib} PARENT_SCOPE)
+set (pnetcdf_incdir ${pnetcdf_incdir} PARENT_SCOPE)
+set (netcdf_c_incdir ${netcdf_c_incdir} PARENT_SCOPE)
+set (netcdf_f_incdir ${netcdf_f_incdir} PARENT_SCOPE)
+
+# Create the interface library, and set target properties
+add_library(netcdf INTERFACE)
+target_link_libraries(netcdf INTERFACE ${pnetcdf_lib};${netcdf_c_lib};${netcdf_f_lib})
+target_include_directories(netcdf INTERFACE ${pnetcdf_incdir};${netcdf_c_incdir};${netcdf_f_incdir})
+if (pnetcdf_lib)
+  target_compile_definitions(netcdf INTERFACE "-D_PNETCDF")
```

Comment:
Is this CPP def needed? I see `_PNETCDF` only inside scorpio, but it is locally defined inside `pio.h` based on other macros. Can we purge this?

Reply:
It's a little tricky to grep for, tons of false positives. I think you are right that it's used by PIO which gets build in the shared lib phase and so doesn't use this code.

Reply:
Also, scorpio sets it internally in pio.h:
```c++
#include "pio_config.h"
#if PIO_USE_PNETCDF
  #define _PNETCDF 1
#endif
```
I think this is one of those macros that were used at some point, but when they became unused nobody took care of removing them (perhaps based on the same fear we all have: "what if it's actually needed somewhere?").

Without grepping in submodules, I don't get any meaningful match of the _whole word_ `_PNETCDF`. Only scorpio has many matches, but again, it does define it internally.

I would consider pruning it.

Reply:
I agree. Pruned.